internal/rbd/controllerserver.go

```diff
@@ -622,7 +622,6 @@ func (cs *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 		}
 		return &csi.DeleteVolumeResponse{}, nil
 	}
```

Comment:
I am not certain it is a extra `Destroy()`. The initial defer is placed on an object that is empty, and the object/pointer is overwritten by a return value of a function. Better would be to do
```golang
var rbdVol *rbdVolume

rbdVol, err = genVolFromVolID(...)
if err != nil {
    ...
}
defer rbdVol.Destroy()
```

This first destroy in the current code might be completely useless.

Reply:
as per the above snippet, the Destroy won't be called in case of an error

Reply:
that is fine, as `genVolFromVolID()` does not return an rbdVolume if an error occurs

Reply:
we already have an rbd connection opened we need to close that one right in case of error also. that why I moved the Destroy right after the declaration and even in case of error also `genVolFromVolID` returns the rbdVolume structure

Reply:
update the Destory call